functorch/experimental/output_dtype.py

```diff
@@ -0,0 +1,136 @@
+from functools import partial 
+
+import torch
+import torch.utils._pytree as pytree
+from torch.fx.experimental.proxy_tensor import (
+    ProxyTorchDispatchMode,
+    disable_proxy_modes_tracing,
+    ProxyTorchDispatchMode,
+    track_tensor_tree,
+    unwrap_proxy,
+)
+from torch.utils._python_dispatch import (
+    _get_current_dispatch_mode,
+    _pop_mode_temporarily,
+)
+from torch._C import DispatchKey
+from torch._functorch.eager_transforms import (
+    _unwrap_all_tensors_from_functional,
+    _wrap_all_tensors_to_functional,
+)
+from torch._ops import HigherOrderOperator
+from torch._subclasses.fake_tensor import FakeTensorMode
+
+
+output_dtype = HigherOrderOperator("output_dtype")
```

Comment:
Let's put the implementation into `torch/_higher_order_ops/output_dtype.py` so we don't need to do the work of moving it out of functorch.experimental later. Dynamo has some interesting behavior on files in the torch namespace and not in the torch namespace.